internal/resources/snapshot_backup.go

```diff
@@ -0,0 +1,620 @@
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var (
+		organizationId = plan.OrganizationId.ValueString()
+		projectId      = plan.ProjectID.ValueString()
+		clusterId      = plan.ClusterID.ValueString()
+	)
+
+	createSnapshotBackupRequest := snapshot_backup.CreateSnapshotBackupRequest{
+		Retention:     int(plan.Retention.ValueInt64()),
+		RegionsToCopy: providerschema.ConvertRegionsToCopy(plan.RegionsToCopy),
+	}
+
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/cloudsnapshotbackups", s.HostURL, organizationId, projectId, clusterId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusAccepted}
+	createResp, err := s.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		createSnapshotBackupRequest,
+		s.Token,
+		nil,
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error executing create snapshot backup",
+			errorMessageWhileSnapshotBackupCreation+api.ParseError(err),
+		)
+		tflog.Debug(ctx, "error executing create snapshot backup", map[string]interface{}{
+			"organizationId":              organizationId,
+			"projectId":                   projectId,
+			"clusterId":                   clusterId,
+			"createSnapshotBackupRequest": createSnapshotBackupRequest,
+			"createResp":                  createResp,
+			"err":                         api.ParseError(err),
+		})
+		return
+	}
+
+	var createSnapshotBackupResponse snapshot_backup.CreateSnapshotBackupResponse
+	err = json.Unmarshal(createResp.Body, &createSnapshotBackupResponse)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error unmarshalling create snapshot backup response",
+			"Could not unmarshal create snapshot backup response: "+err.Error(),
+		)
+		tflog.Debug(ctx, "error unmarshalling create snapshot backup response", map[string]interface{}{
+			"organizationId":              organizationId,
+			"projectId":                   projectId,
+			"clusterId":                   clusterId,
+			"createSnapshotBackupRequest": createSnapshotBackupRequest,
+			"createResp":                  createResp,
+			"err":                         api.ParseError(err),
+		})
+		return
+	}
+
+	// Checks the snapshot backup creation is complete.
+	backupResp, err := s.checkSnapshotBackupStatus(ctx, organizationId, projectId, clusterId, createSnapshotBackupResponse.ID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error while checking latest snapshot backup status",
+			errorMessageWhileSnapshotBackupCreation+api.ParseError(err),
+		)
+		tflog.Debug(ctx, "error checking latest snapshot backup status", map[string]interface{}{
+			"organizationId": organizationId,
+			"projectId":      projectId,
+			"clusterId":      clusterId,
+			"Id":             createSnapshotBackupResponse.ID,
+			"err":            err,
+		})
+		return
+	}
+
+	refreshedState, err := createSnapshotBackup(ctx, backupResp, clusterId, projectId, organizationId)
```

Comment:
nit

function name is a bit confusing.  it reads like its creating a new backup.  change it to something like `morphToTerraformCloudSnapshotBackup`